File: javascript/src/fizzbuzz.js
// returns an array from 1 to max_number with values according to the fizzbuzz rules 

var fizzbuzz = function(max_number) {
	result = []
	for (var i = 1; i < max_number+1; i++) {
		if ((i % 3 == 0) && (i % 5 == 0)) {
			result.push("FizzBuzz");
		} else if (i % 3 == 0) {
			result.push("Fizz");
		} else if (i % 5 == 0) {
			result.push("Buzz");
		} else {
			result.push(i);
		};
	};
    return result;
};

File: ruby/fizzbuzz.rb
require 'test/unit'

# returns an array from 1 to max_number with values according to the fizzbuzz rules 
def fizzbuzz(max_number)
  result = []
  (1..max_number).each do |n|
    if (n % 3 == 0) && (n % 5 == 0)
      result << "FizzBuzz"
    elsif (n % 3 == 0)
      result << "Fizz"
    elsif (n % 5 == 0)
      result << "Buzz"
    else
      result << n
    end
  end
  result
end

###
### Tests
###

class TestFizzBuzz < Test::Unit::TestCase
    def test_fizzbuzz
      assert_equal fizzbuzz(15), [1, 2, "Fizz", 4, "Buzz", "Fizz", 7, 8, "Fizz", "Buzz", 11, "Fizz", 13, 14, "FizzBuzz"]
    end
end